server/server.js

```diff
@@ -20,14 +21,38 @@ var Donation    = require('./models/donation').Donation;
 // CONFIGURE APP
 // =============================================================================
 var app         = express();
-var api         = require ('./routes/api');
-var webhooks    = require('./routes/webhooks');
+
+app.use((req, res, next) => {
```

Comment:
Do you have to put this on app? Can we create an Express Route obj for just the Shopify endpoints and add this crypto step there?

Reply:
Yeah that’s what I wanted to do. It’s technically possible to use route.use() on the webhook router to add the middleware there, but I had a bunch of problems with that. That is because the middleware in Express is called in the order it is declared. And body parser does some dark magic to the request body that changes it from its raw form. I have to use the raw request body exactly how it is sent from Shopify in order to get the correct hash. So this has to be declared on app before bodyParser.json

Reply:
Which is annoying because it messes with the nice little organization we have going and requires the regex conditional below.